awx/main/models/workflow.py

```diff
@@ -82,7 +82,7 @@ def get_parent_nodes(self):
         success_parents = getattr(self, '%ss_success' % self.__class__.__name__.lower()).all()
         failure_parents = getattr(self, '%ss_failure' % self.__class__.__name__.lower()).all()
         always_parents = getattr(self, '%ss_always' % self.__class__.__name__.lower()).all()
-        return success_parents | failure_parents | always_parents
+        return (success_parents | failure_parents | always_parents).order_by('id')
```

Comment:
iiiinteresting, it surprises me that this works with this syntax, but why not? I guess this is what enforces the parental ordering that we've mentioned several times.

Reply:
boolean logic operations on querysets always produce a new queryset object.